### PR TITLE
[wasm] Disable some existing System.Console library tests throwing PNSE

### PR DESCRIPTION
**Pull request by MaximLipnin**
- Disabled the existing System.Console library tests which fail on WASM with PlatformNotSupportedException
- Added several tests to reflect WASM PNSE-behavior.

This is the follow up for https://github.com/dotnet/runtime/pull/37645

cc @steveisok

### CODE REVIEW

File: src/libraries/System.Console/tests/Color.cs [hunk 1]
```diff
@@ -32,6 +34,15 @@ public static void RoundtrippingColor()
     }
 
     [Fact]
+    [PlatformSpecific(TestPlatforms.Browser)]
+    public static void RoundtrippingColor_Browser()
+    {
+        Assert.Throws<PlatformNotSupportedException>(() => Console.BackgroundColor = Console.BackgroundColor );
+        Assert.Throws<PlatformNotSupportedException>(() => Console.ForegroundColor = Console.ForegroundColor );
+    }
```
Comment (stephentoub) [inline src/libraries/System.Console/tests/Color.cs:42]: Is this testing that the setter or the getter throws?  As written, we can't tell.  Assuming both throw, it'd be better to have four separate asserts:
```C#
Assert.Throws<PlatformNotSupportedException>(() => Console.BackgroundColor);
Assert.Throws<PlatformNotSupportedException>(() => Console.ForegroundColor);
Assert.Throws<PlatformNotSupportedException>(() => Console.BackgroundColor = ConsoleColor.Red);
Assert.Throws<PlatformNotSupportedException>(() => Console.ForegroundColor = ConsoleColor.Red);
```
at which point the test should also be renamed appropriately.

File: src/libraries/System.Console/tests/ReadAndWrite.cs [hunk 2]
```diff
@@ -316,6 +316,13 @@ public static void OutputEncoding_Browser()
         Assert.Throws<PlatformNotSupportedException>(() => Console.OutputEncoding = curEncoding );
     }
 
+    [Fact]
+    [PlatformSpecific(TestPlatforms.Browser)]
+    public static void InputEncoding_Browser()
```
Comment (stephentoub) [inline src/libraries/System.Console/tests/ReadAndWrite.cs:321]: I'd use a different suffix than `_Browser`... presumably there will be other platforms (iOS? Android? etc.) where such members throw.  I think it'd be better to just state what's being tested, e.g. `InputEncoding_Throws_PlatformNotSupportedException`, and it's then just enabled on the relevant platforms.

File: src/libraries/System.Console/tests/ReadKey.cs [hunk 3]
```diff
@@ -39,19 +39,33 @@ public static void ConsoleKeyValueCheck()
...
     {
         Assert.Throws<PlatformNotSupportedException>(() => Console.NumberLock);
     }
 
     [Fact]
-    [PlatformSpecific(TestPlatforms.AnyUnix)]
+    [PlatformSpecific(TestPlatforms.Browser)]
+    public void NumberLock_Browser()
+    {
+        Assert.False(Console.NumberLock);
```
Comment (akoeplinger) [inline src/libraries/System.Console/tests/ReadKey.cs:52]: this actually sounds like something we should fix in our Wasm Console implementation. If other Unix implementations already throw PNSE from Console.NumberLock we should do the same on Wasm.

File: src/libraries/System.Console/tests/ReadKey.cs [hunk 4]
```diff
@@ -39,19 +39,33 @@ public static void ConsoleKeyValueCheck()
...
     public void CapsLock_GetUnix_ThrowsPlatformNotSupportedException()
     {
         Assert.Throws<PlatformNotSupportedException>(() => Console.CapsLock);
     }
 
+    [Fact]
+    [PlatformSpecific(TestPlatforms.Browser)]
+    public void CapsLock_Browser()
+    {
+        Assert.False(Console.CapsLock);
```
Comment (akoeplinger) [inline src/libraries/System.Console/tests/ReadKey.cs:66]: Same here.

**Review by akoeplinger — COMMENTED** [review]
Looks good in general, just a comment on NumberLock/CapsLock

File: src/libraries/System.Console/tests/ReadKey.cs (hunk 3 above)
Comment (MaximLipnin) [reply src/libraries/System.Console/tests/ReadKey.cs:52]: We have `public static bool NumberLock => false;` for both [Android ](https://github.com/dotnet/runtime/blob/master/src/libraries/System.Console/src/System/ConsolePal.Android.cs#L50)and [iOS ](https://github.com/dotnet/runtime/blob/master/src/libraries/System.Console/src/System/ConsolePal.iOS.cs#L55)implementations.

File: src/libraries/System.Console/tests/ReadKey.cs (hunk 4 above)
Comment (MaximLipnin) [reply src/libraries/System.Console/tests/ReadKey.cs:66]: We have `public static bool CapsLock => false;` for both [Android ](https://github.com/dotnet/runtime/blob/master/src/libraries/System.Console/src/System/ConsolePal.Android.cs#L52) and [iOS ](https://github.com/dotnet/runtime/blob/master/src/libraries/System.Console/src/System/ConsolePal.iOS.cs#L57)implementations.

File: src/libraries/System.Console/tests/Color.cs (hunk 1 above)
Comment (MaximLipnin) [reply src/libraries/System.Console/tests/Color.cs:50]: Divided into separate asserts

File: src/libraries/System.Console/tests/ReadAndWrite.cs (hunk 2 above)
Comment (MaximLipnin) [reply src/libraries/System.Console/tests/ReadAndWrite.cs]: Made test names more specific

**Review by kjpou1 — APPROVED** [review]

File: src/libraries/System.Console/tests/ReadKey.cs (hunk 3 above)
Comment (akoeplinger) [reply src/libraries/System.Console/tests/ReadKey.cs:52]: Ok, we should change these implementations then to be aligned with the other Unix implementations. We can do that in a follow-up PR.

File: src/libraries/System.Console/tests/WindowAndCursorProps.cs [hunk 5]
```diff
@@ -112,23 +112,37 @@ public static void WindowHeight_SetUnix_ThrowsPlatformNotSupportedException()
...
+    public static void WindowWidth_Getter_Throws_PlatformNotSupportedException()
+    {
+        Assert.Throws<PlatformNotSupportedException>(() => Console.WindowWidth);
+    }
+
+    [Fact]
+    [PlatformSpecific(TestPlatforms.AnyUnix & ~TestPlatforms.Browser)]  // Expected behavior specific to Unix
     public static void WindowLeft_GetUnix_ReturnsZero()
     {
         Assert.Equal(0, Console.WindowLeft);
```
Comment (akoeplinger) [inline src/libraries/System.Console/tests/WindowAndCursorProps.cs:148]: this is also something where we should align our iOS/Android/WASM with the other Unix implementations.

**Review by akoeplinger — APPROVED** [review]
Looks good with some follow-up.

**Pull request merged**